Add Publishing Checklist to appropriate Manage Post screens

davisshaver opened the pull request:
See https://github.com/fusioneng/publishing-checklist/issues/4

danielbachhuber commented on publishing-checklist.php:
```diff
@@ -116,6 +118,41 @@ private function get_template_part( $template, $vars = array() ) {
+
+			if ( empty( $this->tasks ) ) {
+				return;
+			}
```
To make our code more DRY, can we have some method to actually evaluate the checklist for the post, and use it in both places? I'm happy to put together a PR to show you what I'm talking about.

danielbachhuber commented:
Can you share a screenshot?

davisshaver commented:
@danielbachhuber 
![jun 23 2015 09 11](https://cloud.githubusercontent.com/assets/1636964/8306688/f0600022-1987-11e5-9d10-f42b2053992e.gif)

davisshaver commented:
- Slimmed down version of Edit UI.
- Refactored JS to work with multiple checklists on a page.
- Added test to ensure that manage posts column has checklist.

davisshaver commented:
@danielbachhuber could you review?

danielbachhuber commented on publishing-checklist.php:
```diff
@@ -155,6 +152,27 @@ private function get_template_part( $template, $vars = array() ) {
+	 * Customize columns on the "Manage Posts" views
+	 */
+	public function filter_manage_posts_columns( $columns ) {
+		$columns['checklist'] = esc_html__( 'Checklist', 'fusion', 'publishing-checklist' );
```
Can we slug this as `publishing_checklist`, and call the column "Publishing Checklist"?

You can drop the `'fusion'`

danielbachhuber commented on publishing-checklist.php:
```diff
@@ -155,6 +152,27 @@ private function get_template_part( $template, $vars = array() ) {
+	 * Handle the output for a custom column
+	 */
+	public function action_manage_posts_custom_column( $column_name, $post_id ) {
+		if ( 'checklist' === $column_name ) {
```
Once above is addressed, this will need to be updated too.

danielbachhuber commented on publishing-checklist.php:
```diff
@@ -22,12 +22,7 @@ public static function get_instance() {
-
-			if ( defined( 'WP_CLI' ) && WP_CLI ) {
-				require_once dirname( __FILE__ ) . '/inc/class-cli-command.php';
-			}
```
Why are you removing this?

danielbachhuber commented:
Like Quick Actions, can we make "Show List" hidden by default, and then appear on hover?

![image](https://cloud.githubusercontent.com/assets/36432/8352896/9d02c1a0-1aed-11e5-8b38-eecdf7a6a4dc.png)

danielbachhuber commented on assets/js/src/publishing-checklist.js:
```diff
@@ -1,17 +1,15 @@
-	showButton.on('click', function(){
-		showButton.hide();
-		checklistItems.show();
+	$(document).on('click', '.publishing-checklist-show-list', function() {
```
Generally, it's inadvisable to bind a click event to `$(document)` because the event will fire for every click on the document, and perform calculations to determine whether this callback should be fired.

Can you bind it to the `.publishing-checklist-show-list` element instead? You'll be able to do the same traversal you have.

```
$('.publishing-checklist-show-list').on('click'
```

However, the event may need to be rebound when Quick Edit is used. If this is the case, you could bind the event like so:

```
$('.wp-list-table').on('click','.publishing-checklist-show-list'
```

davisshaver commented:
![now](https://cloud.githubusercontent.com/assets/1636964/8354826/bbda98b4-1b10-11e5-9ad4-11a847c96799.gif)
`Show List` will now only appear on row hover. I used the existing `.row-actions` helper to accomplish this.

davisshaver commented:
@danielbachhuber review/merge?

danielbachhuber commented:
:+1:
